src/pinyin.cc

```diff
@@ -18,14 +18,14 @@ std::set<std::string> PinYin::to_plain(const std::string &input) {
   std::set<std::string> s;
   std::string value;
   for (size_t i = 0, len = 0; i != input.length(); i += len) {
-    unsigned char byte = (unsigned char)input[i];
+    auto byte = input[i];
     if (byte == ',') {
       s.insert(value);
       s.insert(value.substr(0, 1));
       value.clear();
       continue;
     }
-    len = get_str_len(byte);
+    len = get_str_len((unsigned char)byte);
     if (len == 1) {
       value.push_back(byte);
```

Comment:
Clang-Tidy: Narrowing conversion from 'unsigned char' to signed type 'char' is implementation-defined